app/lib/theme/schemaFilteringHelpers.ts

```diff
@@ -12,18 +12,27 @@ export const getSchemaAndDataIncludingCalculatedValues = (
 ) => {
   const filteredSchema = {
     ...JSON.parse(JSON.stringify(formSchema)),
-    ...additionalSchemaProperties,
+    properties: {
+      ...JSON.parse(JSON.stringify(formSchema)).properties,
+      ...additionalSchemaProperties,
+    },
   };
 
   const newDataObject = {};
 
   for (const key of Object.keys(filteredSchema.properties)) {
+    // null new data with undefined old data occurs when an optional form (e.g. budgets) is created but not filled
```

Comment:
This is what causes the bug where we see labels but no diffs: 
![Screenshot from 2023-06-27 15-18-19](https://github.com/bcgov/cas-cif/assets/77306764/d45e29af-9ca9-47d1-8eb3-8faa8db4c895)
We could also handle this in `CustomDiffs` and I can make a tech debt card if people think that would be a better approach. The milestone form uses a custom schema filtering helper, so it might have the same bug

Reply:
That makes sense to me. I think I faced that issue before when I was implementing the schema filtering function for the Emissions Intensity Report.